Mathlib/Analysis/CStarAlgebra/CStarMatrix.lean

```diff
@@ -0,0 +1,680 @@
+  map_add' M₁ M₂ := by
+    ext
+    simp only [ContinuousLinearMap.coe_mk', LinearMap.coe_mk, AddHom.coe_mk, Function.comp_apply,
+      WithCStarModule.equivL_apply, WithCStarModule.equivL_symm_apply,
+      WithCStarModule.equiv_symm_pi_apply, ContinuousLinearMap.add_apply, WithCStarModule.add_apply]
+    rw [Matrix.add_mulVec, Pi.add_apply]
+  map_smul' c M := by
+    ext x i
+    simp only [ContinuousLinearMap.coe_mk', LinearMap.coe_mk, AddHom.coe_mk, Function.comp_apply,
+      WithCStarModule.equivL_apply, WithCStarModule.equivL_symm_apply,
+      WithCStarModule.equiv_symm_pi_apply, Matrix.mulVec, dotProduct,
+      WithCStarModule.equiv_pi_apply, RingHom.id_apply, ContinuousLinearMap.coe_smul',
+      Pi.smul_apply, WithCStarModule.smul_apply, Finset.smul_sum]
+    congr
+    ext j
+    rw [CStarMatrix.smul_apply, smul_mul_assoc]
+
+lemma toCLM_apply {M : CStarMatrix m n A} {v : C⋆ᵐᵒᵈ (n → A)} :
+    toCLM M v = (WithCStarModule.equiv _).symm (M.mulVec v) := rfl
+
+lemma toCLM_apply_eq_sum {M : CStarMatrix m n A} {v : C⋆ᵐᵒᵈ (n → A)} :
+    toCLM M v = (WithCStarModule.equiv _).symm (fun i => ∑ j, M i j * v j) := by
+  ext i
+  simp [toCLM_apply, Matrix.mulVec, dotProduct]
+
+/-- Interpret a `CStarMatrix m n A` as a continuous linear map acting on `C⋆ᵐᵒᵈ (n → A)`. This
+version is specialized to the case `m = n` and is bundled as a non-unital algebra homomorphism. -/
+def toCLMNonUnitalAlgHom : CStarMatrix n n A →ₙₐ[ℂ] (C⋆ᵐᵒᵈ (n → A)) →L[ℂ] (C⋆ᵐᵒᵈ (n → A)) :=
+  { toCLM (n := n) (m := n) with
+    map_zero' := by ext1; simp
+    map_mul' := by intros; ext; simp [toCLM] }
+
+lemma toCLMNonUnitalAlgHom_eq_toCLM {M : CStarMatrix n n A} :
+    toCLMNonUnitalAlgHom (A := A) M = toCLM M := rfl
+
+open WithCStarModule in
+@[simp high]
+lemma toCLM_apply_single [DecidableEq n] {M : CStarMatrix m n A} {j : n} (a : A) :
+    (toCLM M) (equiv _ |>.symm <| Pi.single j a) = (equiv _).symm (fun i => M i j * a) := by
+  simp [toCLM]
+
+open WithCStarModule in
+lemma toCLM_apply_single_apply [DecidableEq n] {M : CStarMatrix m n A} {i : m} {j : n} (a : A) :
+    (toCLM M) (equiv _ |>.symm <| Pi.single j a) i = M i j * a := by simp
+
+open WithCStarModule in
+lemma mul_entry_mul_eq_inner_toCLM [DecidableEq m] [DecidableEq n] {M : CStarMatrix m n A}
+    {i : m} {j : n} (a b : A) :
+    star a * M i j * b
+      = ⟪equiv _ |>.symm (Pi.single i a), toCLM M (equiv _ |>.symm <| Pi.single j b)⟫_A := by
+  simp [mul_assoc, inner_def]
+
+lemma toCLM_injective [DecidableEq n] : Function.Injective (toCLM (A := A) (m := m) (n := n)) := by
+  rw [injective_iff_map_eq_zero]
+  intro M h
+  ext i j
+  rw [Matrix.zero_apply, ← norm_eq_zero, ← sq_eq_zero_iff, sq, ← CStarRing.norm_self_mul_star,
+    ← toCLM_apply_single_apply]
+  simp [h]
+
+open WithCStarModule in
+lemma toCLM_inner_right_eq_left {M : CStarMatrix m n A} {v : C⋆ᵐᵒᵈ (m → A)}
+    {w : C⋆ᵐᵒᵈ (n → A)} : ⟪v, toCLM M w⟫_A = ⟪toCLM Mᴴ v, w⟫_A := by
```

Comment:
Shouldn't this be called:
```suggestion
lemma inner_toCLM_right {M : CStarMatrix m n A} {v : C⋆ᵐᵒᵈ (m → A)}
    {w : C⋆ᵐᵒᵈ (n → A)} : ⟪v, toCLM M w⟫_A = ⟪toCLM Mᴴ v, w⟫_A := by
```
Or maybe the direction should be reversed and it should called `inner_toCLM_conjTranspose_left`.

Reply:
I went for the second suggestion.